pkg/component/networking/istio/istiod.go

```diff
@@ -251,54 +251,57 @@ func (i *istiod) Deploy(ctx context.Context) error {
 	}
 
 	registry := managedresources.NewRegistry(kubernetes.SeedScheme, kubernetes.SeedCodec, kubernetes.SeedSerializer)
-	if err := registry.Add(&monitoringv1.ServiceMonitor{
-		ObjectMeta: monitoringutils.ConfigObjectMeta("istio-ingressgateway", v1beta1constants.IstioSystemNamespace, aggregate.Label),
-		Spec: monitoringv1.ServiceMonitorSpec{
-			Selector:          metav1.LabelSelector{MatchLabels: map[string]string{v1beta1constants.LabelApp: "istio-ingressgateway"}},
-			NamespaceSelector: monitoringv1.NamespaceSelector{Any: true},
-			Endpoints: []monitoringv1.Endpoint{{
-				Path: "/stats/prometheus",
-				Port: "tls-tunnel",
-				RelabelConfigs: []monitoringv1.RelabelConfig{{
-					SourceLabels: []monitoringv1.LabelName{"__meta_kubernetes_pod_ip"},
-					Action:       "replace",
-					TargetLabel:  "__address__",
-					Regex:        `(.+)`,
-					Replacement:  ptr.To("${1}:15020"),
+	// Only add a ServiceMonitor for the istio-ingressgateway if istio is deployed on a seed.
```

Comment:
Can you add an explanation why this is done?

Reply:
Addressed in https://github.com/gardener/gardener/pull/10143/commits/7eab172661b97efe57189763e38ea1e42617e667. PTAL whether this is what you wanted.